packages/xo-web/src/xo-app/proxies/deploy-proxy.js

```diff
@@ -249,20 +255,31 @@ const deployProxy = async ({ proxy } = {}) => {
     )
   }
 
-  const header = (
-    <span>
-      <Icon icon='proxy' />{' '}
-      {isRedeployMode ? _('redeployProxy') : _('deployProxy')}
-    </span>
-  )
-
+  const title = isRedeployMode ? _('redeployProxy') : _('deployProxy')
   if (license === undefined) {
-    return alert(
-      header,
-      <div className='text-muted'>
-        <Icon icon='info' /> {_('noLicenseAvailable')}
-      </div>
-    )
+    const value = 'trial'
+    const choice = await chooseAction({
+      body: (
+        <div className='text-muted'>
+          <Icon icon='info' /> {_('noLicenseAvailable')}
+        </div>
+      ),
+      buttons: [
+        {
+          btnStyle: 'success',
+          icon: 'trial',
+          label: _('trialStartButton'),
+          value,
+        },
+      ],
+      icon: 'proxy',
+      title,
+    }).catch(noop)
+    if (choice !== value) {
+      return
+    }
+
+    license = await createProxyTrialLicense()
```

Comment:
This may also throw with an error like "No trials available". Should we keep an `alert` modal for this?

Reply:
Do you mean an alert displaying this error instead of the `ActionButton` feedback?

Reply:
Yes, what do you think? The same way we had an `alert` when we couldn't find a license. This time, the user can go one step further and request a trial but if that also fails, we could show a message.